crates/compiler/src/obj_gen/mod.rs

```diff
@@ -1316,7 +1314,8 @@ impl ObjectGenerator {
         // offsets.
         let mut accumulated_offset = initial_offset;
 
-        for (elem_ty, elem_val) in struct_elements.iter().zip(element_variables.into_iter()) {
+        let elems_with_vars_and_ix = struct_elements.iter().zip(element_variables).enumerate();
```

Comment:
Hmm. We now have the option of simplifying this to something like a memcpy -- moving whole felts at once. Do we want to note this as an optimization point, or perhaps think about introducing aligned and unaligned memcpy polyfills?

Reply:
I've added a note with a link to #126 so that it does not stay _just_ in the code.